src/core/component.js

```diff
@@ -136,6 +126,38 @@ Component.prototype = {
     return utils.extend({}, data);
   },
 
+  updateCachedValue: function (value, isSinglePropSchema) {
+    var parsedValue;
+    if (value === '') {
+      this.attrValue = undefined;
+      return;
+    }
+    if (typeof value === 'string') {
+      if (isSingleProp(this.schema)) {
+        parsedValue = this.schema.parse(value);
+        // To avoid bogus double parsings. Like for example
+        // when parsing a src id to it's url #monster -> models/monster.dae
+        if (typeof parsedValue === 'string') { parsedValue = value; }
+        this.attrValue = parsedValue;
+      } else {
+        // We just parse using the style parser to avoid double parsing
+        // of individual properties.
+        this.attrValue = styleParser.parse(value);
+      }
+      return;
+    }
+    this.attrValue = value !== undefined ? extendProperties({}, value, isSinglePropSchema) : this.attrValue;
+  },
+
+  /**
+   * Writes cached attribute data to the entity DOM element.
```

Comment:
Double note: we should portray it as "component value" rather than "cached attribute data" to signify what the source of truth is